api/agent/stats.go

```diff
@@ -161,16 +184,22 @@ func IncrementTooBusy(ctx context.Context) {
 }
 
 func (s *stats) Stats() Stats {
+	// this creates a Stats from a stats
+	// stats is the internal struc which is continuously updated, and access is controlled using its Mutex
+	// Stats is a deep copy for external use, and can be converted to JSON
 	var stats Stats
 	s.mu.Lock()
 	stats.Running = s.running
 	stats.Complete = s.complete
 	stats.Queue = s.queue
 	stats.Failed = s.failed
-	stats.FunctionStatsMap = make(map[string]*FunctionStats)
-	for key, value := range s.functionStatsMap {
-		thisFunctionStats := &FunctionStats{Queue: value.queue, Running: value.running, Complete: value.complete, Failed: value.failed}
-		stats.FunctionStatsMap[key] = thisFunctionStats
+	stats.Apps = make(map[string]AppStats)
```

Comment:
not a big deal, but this make statement doesn't have in locked scope.

Reply:
Noted, thanks.